guides/bulk.md

```diff
@@ -0,0 +1,209 @@
+        .Id(2)
+        .Document(new { Title = "Beauty and the Beast - Live Action", Year = 2017 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Id(1)
+        .Document(new { Title = "The Lion King", Year = 1994 })
+    ));
+```
+
+As you can see, each bulk operation is comprised of two objects. The first object contains the operation type and the target document's `_index` and `_id`. The second object contains the document's data. As a result, the body of the request above contains six objects for three index actions.
+
+Alternatively, the `bulk` method can accept an array of hashes where each hash represents a single operation. The following code is equivalent to the previous example:
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Index<object>(i => i
+        .Index(movies)
+        .Id(1)
+        .Document(new { Title = "Beauty and the Beast", Year = 1991 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Id(2)
+        .Document(new { Title = "Beauty and the Beast - Live Action", Year = 2017 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Id(1)
+        .Document(new { Title = "The Lion King", Year = 1994 })
+    ));
+```
+
+We will use this format for the rest of the examples in this guide.
+
+### Creating multiple documents
+
+Similarly, instead of calling the `create` method for each document, you can use the `bulk` API to create multiple documents in a single request. The following code creates three documents in the `movies` index and one in the `books` index:
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 2", Year = 2030 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 3", Year = 2031 })
+    )
+    .Index<object>(i => i
+        .Index(movies)
+        .Document(new { Title = "Beauty and the Beast 4", Year = 2049 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Document(new { Title = "The Lion King 2", Year = 1998 })
+    )
+    .Index<object>(i => i
+        .Index(books)
+        .Document(new { Title = "The Lion King 2", Year = 1998 })
+    ));
+```
+
+Note that we specified only the `_index` for the last document in the request body. This is because the `bulk` method accepts an `index` parameter that specifies the default `_index` for all bulk operations in the request body. Moreover, we omit the `_id` for each document and let OpenSearch generate them for us in this example, just like we can with the `create` method.
+
+### Updating multiple documents
+
+```cs
+var response = await client.BulkAsync(b => b
+    .Update<object>(i => i
+        .Index(movies)
+        .Id(1)
+        .Document(new { Year = 1992 })
+    )
+    .Update<object>(i => i
+        .Index(movies)
+        .Id(2)
+        .Document(new { Year = 2018 })
+    ));
```

Comment:
This is supposed to use a "global" index, not individual, additionally the method for update bodies is `Doc` not `Document`, like so:
```suggestion
var response = await client.BulkAsync(b => b
    .Index(movies)
    .Update<object>(i => i
        .Id(1)
        .Doc(new { Year = 1992 })
    )
    .Update<object>(i => i
        .Id(2)
        .Doc(new { Year = 2018 })
    ));
```